File: Algorithms/cpp/190 - Reverse Bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        // var
        uint32_t result = 0;
        uint32_t exp = INT_MAX + 1;
        
        while (n) {
            result += n % 2 ? exp : 0;
            exp >>= 1;
            n >>= 1;
        }
        return result;
    }
};

File: python/190 - Reverse Bits.py
class Solution(object):
    def reverseBits(self, n):
        """
        :type n: int
        :rtype: int
        """
        ans = 0
        exp = 2**31
        while n > 0:
            ans += (n % 2) * exp
            exp /= 2
            n /= 2
        return ans